rockcraft.yaml

```diff
@@ -3,36 +3,89 @@ base: ubuntu:22.04 # the base environment for this ROCK
 version: '8.0.31' # just for humans. Semantic versioning is recommended
 summary: Charmed MySQL ROCK OCI # 79 char long summary
 description: |
-    MySQL built from the official ubuntu MySQL package 
-    from the Ubuntu repository and further installs 
+    MySQL built from the official MySQL package 
+    from the MySQL repository and installs 
     mysql-shell.  For more information on ROCKs, visit 
     the https://github.com/canonical/rockcraft.
 license: Apache-2.0 # your application's SPDX license
-cmd: 
+cmd:
     - /usr/bin/setpriv
     - --clear-groups
     - --reuid
     - mysql
     - --regid
     - mysql
-    - -- 
+    - --
     - /usr/sbin/mysqld
 platforms: # The platforms this ROCK should be built on and run on
     amd64:
 
 parts:
-    mysql:
+    mysql-repo:
         plugin: nil
+        override-pull: |
+            export DEBIAN_FRONTEND=noninteractive
+            export CONFIG_VERSION="0.8.22-1"
+            apt-get update
+            apt-get install -y wget lsb-release gnupg
+            wget -q https://dev.mysql.com/get/mysql-apt-config_${CONFIG_VERSION}_all.deb \
+                -O /tmp/mysql-apt-config_${CONFIG_VERSION}_all.deb
+            dpkg -i /tmp/mysql-apt-config_${CONFIG_VERSION}_all.deb
+            apt-get update
+    percona-repo:
+        plugin: nil
+        override-pull: |
+            export DEBIAN_FRONTEND=noninteractive
+            apt-get update
+            apt-get install -y wget lsb-release
+            wget https://repo.percona.com/apt/percona-release_latest.$(lsb_release -sc)_all.deb
+            dpkg -i percona-release_latest.$(lsb_release -sc)_all.deb
+            apt-get update
+    packages-deb:
+        plugin: nil
+        after:
+            - mysql-repo
+            - percona-repo
         stage-packages:
-            - mysql-server-8.0
-            - util-linux
-        stage-snaps:
-            - xtrabackup/latest/edge
+            - mysql-server
+            - mysql-router
             - mysql-shell
+            - percona-xtrabackup-80
+            - util-linux
+        override-stage: |
+            craftctl default
+            mkdir -p $CRAFT_STAGE/var/lib/mysql
+            ln -sf /root/parts/packages-deb/install/usr/lib/x86_64-linux-gnu/libnuma.so.1 /usr/lib/x86_64-linux-gnu/libnuma.so.1
+            ln -sf /root/parts/packages-deb/install/usr/lib/x86_64-linux-gnu/libaio.so.1 /usr/lib/x86_64-linux-gnu/libaio.so.1
+            rm -rf /var/lib/mysql/
+            mysqld --initialize
+            cp -r /var/lib/mysql $CRAFT_STAGE/var/lib
```

Comment:
I don't believe that works because at the time of creation, the `mysql` user does not exist and so ownership has to be set to a user id which will eventually be the id for the `mysql` user that is created later.  

I could be wrong about that, but from my understanding, that would be an issue.